File: InstantSearch/app/src/main/java/com/woowrale/architecture/mvppatterns/ui/search/SearchActivity.java
package com.woowrale.architecture.mvppatterns.ui.search;

import android.content.Intent;
import android.graphics.Color;
import android.os.Build;
import android.os.Bundle;
import android.support.v7.widget.Toolbar;
import android.view.View;

import com.woowrale.architecture.mvppatterns.R;
import com.woowrale.architecture.mvppatterns.ui.base.BaseActivity;
import com.woowrale.architecture.mvppatterns.ui.search.local.LocalSearchActivity;
import com.woowrale.architecture.mvppatterns.ui.search.remote.RemoteSearchActivity;

import butterknife.BindView;
import butterknife.ButterKnife;
import butterknife.OnClick;

public class SearchActivity extends BaseActivity implements SearchView {

    private SearchPresenter presenter;

    @BindView(R.id.toolbar)
    Toolbar toolbar;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_search);
        ButterKnife.bind(this);

        setSupportActionBar(toolbar);
        whiteNotificationBar(toolbar);

        presenter = new SearchPresenter(this);
    }

    @OnClick(R.id.btn_local_search)
    public void openLocalSearch() {
        presenter.navigateToSearch("L");
    }

    @OnClick(R.id.btn_remote_search)
    public void openRemoteSearch() {
        presenter.navigateToSearch("R");
    }

    private void whiteNotificationBar(View view) {
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
            int flags = view.getSystemUiVisibility();
            flags |= View.SYSTEM_UI_FLAG_LIGHT_STATUS_BAR;
            view.setSystemUiVisibility(flags);
            getWindow().setStatusBarColor(Color.WHITE);
        }
    }

    @Override
    public void navigateToSearchLocal() {
        startActivity(new Intent(SearchActivity.this, LocalSearchActivity.class));
    }

    @Override
    public void navigateToSearchRemote() {
        startActivity(new Intent(SearchActivity.this, RemoteSearchActivity.class));
    }
}

File: KInstantSearch/app/src/main/java/com/woowrale/mvp/ui/search/home/SearchActivity.kt
package com.woowrale.mvp.ui.search.home

import android.content.Intent
import android.graphics.Color
import android.os.Build
import android.os.Bundle
import android.view.View
import com.woowrale.mvp.R
import com.woowrale.mvp.ui.base.BaseActivity
import com.woowrale.mvp.ui.search.local.LocalSearchActivity
import com.woowrale.mvp.ui.search.remote.RemoteSearchActivity
import kotlinx.android.synthetic.main.activity_search.*
import kotlinx.android.synthetic.main.content_search.*

class SearchActivity : BaseActivity(),
    SearchView {

    private var presenter: SearchPresenter? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_search)

        setSupportActionBar(toolbar)
        whiteNotificationBar(toolbar)

        presenter = SearchPresenter(this)
        btnLocalSearch.setOnClickListener {
            presenter!!.navigateToSearch("L")
        }

        btnRemoteSearch.setOnClickListener {
            presenter!!.navigateToSearch("R")
        }
    }

    private fun whiteNotificationBar(view: View?) {
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
            var flags = view!!.systemUiVisibility
            flags = flags or View.SYSTEM_UI_FLAG_LIGHT_STATUS_BAR
            view.systemUiVisibility = flags
            getWindow().setStatusBarColor(Color.WHITE)
        }
    }

    override fun navigateToSearchLocal() {
        startActivity(Intent(this@SearchActivity, LocalSearchActivity::class.java))
    }

    override fun navigateToSearchRemote() {
        startActivity(Intent(this@SearchActivity, RemoteSearchActivity::class.java))
    }
}